ui/src/main/java/com/wireguard/android/fragment/TunnelListFragment.kt

```diff
@@ -52,7 +54,20 @@ class TunnelListFragment : BaseFragment() {
         val activity = activity ?: return@registerForActivityResult
         val contentResolver = activity.contentResolver ?: return@registerForActivityResult
         activity.lifecycleScope.launch {
-            TunnelImporter.importTunnel(contentResolver, data) { showSnackbar(it) }
+            if (contentResolver.getType(data)?.startsWith("image/") == true) {
+                val qrCodeFromFileScanner = QrCodeFromFileScanner(contentResolver, QRCodeReader())
+                val qrResult = qrCodeFromFileScanner.scan(data)
```

Comment:
Can we push this logic down to `TunnelImporter`?

Reply:
Ideally yes, maybe, but right now `tunnelFileImportResultLauncher` only calls `importTunnel` with `contentResolver`, however for QR flow, we will need to show the input name dialog (same as `qrImportResultLauncher` which calls `importTunnel` with `parentFragmentManager`)

I don't really want to make `importTunnel(contentResolver: ContentResolver, uri: Uri, messageCallback: (CharSequence) -> Unit)` to require to pass the fragment manager so for QR use case we can show a dialog. (Seems like a bad trade-off to me)

Should it be ok if I move the check `contentResolver.getType(data)?.startsWith("image/") == true` into `QrCodeFromFileScanner` smt like: `fun validContentType(data): Bool` ?

Reply:
That should be fine :+1:

Reply:
Done